solace/src/main/java/org/apache/apex/malhar/solace/AbstractSolaceBaseInputOperator.java

```diff
@@ -0,0 +1,319 @@
+    channelProperties.setReconnectRetries(Integer.parseInt(this.reconnectRetries));
+
+    reconnectRetryMillis = channelProperties.getReconnectRetryWaitInMillis();
+
+
+    try {
+      session = factory.createSession(this.properties, null, new PrintingSessionEventHandler());
+    } catch (JCSMPException e) {
+      DTThrowable.rethrow(e);
+    }
+
+    //logger.debug("Properties Raw: \n{}", properties.toProperties());
+    logger.debug("Properties:\n" + properties.toString());
+    //logger.debug("\n===============================================\n");
+
+    idempotentStorageManager.setup(context);
+    lastCompletedWId = idempotentStorageManager.getLargestRecoveryWindow();
+    //logger.debug("++++++++++++++++++++Largest Completed: " + lastCompletedWId);
+
+
+  }
+
+  @Override
+  public void checkpointed(long arg0)
+  {
+    // TODO Auto-generated method stub
+
+  }
+
+  @Override
+  public void committed(long window)
+  {
+    try {
+      idempotentStorageManager.deleteUpTo(operatorId, window);
+    } catch (IOException e) {
+      DTThrowable.rethrow(e);
+    }
+
+  }
+
+  protected T processMessage(BytesXMLMessage message)
+  {
+    T tuple = convert(message);
+    if (tuple != null) {
+      emitTuple(tuple);
+    }
+    return tuple;
+  }
+
+
+  @Override
+  public void activate(Context.OperatorContext context)
+  {
+    try {
+      session.connect();
+      reliableConsumer = session.getMessageConsumer(rcHandler, cbHandler);
+      //consumer = getConsumer();
+      reliableConsumer.start();
+
+    } catch (JCSMPException e) {
+      DTThrowable.rethrow(e);
+    }
+  }
+
+  @Override
+  public void deactivate()
+  {
+    try {
+      consumer.stop();
+      clearConsumer();
+      consumer.close();
+      reliableConsumer.close();
+    } catch (JCSMPException e) {
+      DTThrowable.rethrow(e);
+    }
+  }
+
+  @Override
+  public void teardown()
+  {
+
+    idempotentStorageManager.teardown();
+    session.closeSession();
+  }
+
+  @Override
+  public void beginWindow(long windowId)
+  {
+    super.beginWindow(windowId);
+    this.windowId = windowId;
+  }
+
+
+  protected abstract T convert(BytesXMLMessage message);
+
+  protected abstract void emitTuple(T tuple);
+
+  protected abstract Consumer getConsumer() throws JCSMPException;
+
+  protected abstract void clearConsumer() throws JCSMPException;
+
+  public void setProperties(JCSMPProperties properties)
+  {
+    this.properties = properties;
+
+  }
+
+
+  public IdempotentStorageManager getIdempotentStorageManager()
+  {
+    return idempotentStorageManager;
+  }
+
+  public void setUnackedMessageLimit(String unackedMessageLimit)
+  {
+    this.unackedMessageLimit = unackedMessageLimit;
+  }
+
+  public String getUnackedMessageLimit()
+  {
+    return unackedMessageLimit;
+  }
+
+  public void setConnectRetries(String connectRetries)
+  {
+    this.connectRetries = connectRetries;
+    logger.info("+++++++++++++++++++reconnectRetries: {}", this.connectRetries);
```

Comment:
Do we want to keep this logger.info in production code? Why is showing this more important than other properties?